ModularTegustation/delivery_doors.dm

```diff
@@ -0,0 +1,95 @@
+// Doors for parcels to be delivered to. Simulated mailman.
+// A good way to get money without combat.
+/obj/structure/delivery_door
+	name = "locked door"
+	desc = "A doorway to somewhere your not allowed to be."
+	icon = 'icons/obj/doors/airlocks/highsec/highsec.dmi'
+	icon_state = "closed"
+	anchored = TRUE
+	layer = CLOSED_DOOR_LAYER
+	var/address = "000"
+	var/list/item_order = list()
+
+/obj/structure/delivery_door/Initialize()
+	. = ..()
+	address = "[x]-[y]"
+	name += " ([address])"
+
+/obj/structure/delivery_door/attackby(obj/item/I, mob/user)
+	var/ordered_item = locate(I) in item_order
+	if(ordered_item)
+		item_order -= ordered_item
+		Reward(user, I, 30 + rand(-1,20))
+		return
+	if(istype(I, /obj/item/delivery_parcel))
+		// Deliver the item.
+		var/obj/item/delivery_parcel/D = I
+		if(D.address == address)
+			Reward(user, D, 150 + rand(-1,20))
+		return
+	return ..()
+
+// Create parcel to be delivered.
+/obj/structure/delivery_door/proc/OrderParcel(origin)
+	if(!isturf(origin) && !isatom(origin))
+		return FALSE
+	var/obj/item/delivery_parcel/D = new (get_turf(origin))
+	D.labelParcel(address)
+	return TRUE
+
+// Order items that are not safety sealed.
+/obj/structure/delivery_door/proc/OrderItems(origin, obj/item/T = /obj/item/food/pizza/margherita)
+	if(!isturf(origin) && !isatom(origin))
+		return FALSE
+	item_order = list(T)
+	var/obj/item/paper/P = new (get_turf(origin))
+	P.setText("<center><b>[address] orders a [initial(T.name)].</b></center>")
+	return TRUE
+
+// Pay the pizzaman
+/obj/structure/delivery_door/proc/Reward(mob/living/user, obj/item/delivery, amt)
+	var/obj/item/holochip/H = new (get_turf(user), amt)
+	user.put_in_hands(H)
+	to_chat(user, span_notice("The parcel is taken and payment is quickly tossed into your hand before the door locks again."))
+	qdel(delivery)
+
+// Delivery Object
+/obj/item/delivery_parcel
+	name = "delivery parcel"
+	desc = "Your attempts to open the package are foiled by a J corp security lock."
```

Comment:
This feels more like a special interaction if you used the item in your hand rather than a description to me, maybe just say that it has a J-corp lock instead of being so upfront

Reply:
Gotcha Gotcha